README.md

```diff
@@ -97,14 +97,43 @@ contractFunction = (state, action) => state;
 
 This function takes a state and an action as input arguments - and produces a new state as a result.
 In functional programming terms it acts as a `fold` function - the state of the contract is derived from:
+
 1. An initial state
 2. A contract function
 3. An ordered list of actions
 
 In order to evaluate contract state, SmartWeave Protocol client:
+
 1. Loads all the contract's interaction transactions up to the requested block height.
-2. Sorts the interaction transactions. The order of the interactions is determined firstly by interaction 
-transaction block height (i.e. when the transaction was mined in the chain) and secondly by `sha256(transactionId + blockHash)`.
-The full ordering is `[ block_height, sha256(transactionId + blockHash) ]`.
+2. Sorts the interaction transactions. The order of the interactions is determined firstly by interaction
+   transaction block height (i.e. when the transaction was mined in the chain) and secondly by `sha256(transactionId + blockHash)`.
+   The full ordering is `[ block_height, sha256(transactionId + blockHash) ]`.
 3. Applies the sorted interactions onto the contract's `handler` function - evaluating contract's state
    up to the requested block height.
+
+### Evolve
+
+Evolving contract is a way of updating contract source. In order to properly evolve the contract following steps need to be reproduced:
+
+1. `canEvolve` property set to `true` MUST be added to the initial state.
+2. `evolve` interaction MUST be added to the original contract source, it MUST set the `evolve` property in the state to `action.input.value` - new contract source transaction id.
+3. `evolve` interaction MUST be conditioned to the owner of the contract.
```

Comment:
I think we should leave this up to the developer, which is why we would use `canEvolve` so that the developer can handle when it's allowed to evolve.

Reply:
@ppedziwiatr So the state already has a property called `canEvolve`. If `canEvolve` is false and there's `evolve` present, then it doesn't evolve.  The developer should programatically set `canEvolve`  to true or false during the evaluation of the interaction so that the executor can know whether the evolve requested was allowed.

Reply:
Hmm, since `canEvolve` is a part of the contract's state, it should be IMHO only modified by interacting with the contract - i.e. by creating an interaction transaction with a contract - so that anyone, who's willing to evaluate the state, will get the exactly same result.

"The developer should programatically set canEvolve to true or false during the evaluation of the interaction" - how exactly this "programatically setting" would work in this case..? You would like to change the value of this flag without interacting with the contract?

Sorry for all the questions, I just want to fully understand your idea.